### PR TITLE
Add byte buffer

### PR DESCRIPTION
**Pull request by TheNeikos**
This adds a bpp method to `ws_buffer`

### CODE REVIEW

**Comment by neithernut** [comment]
Mh... one of us has to implement the bpp getter for `wl_wayland_buffer`s

File: src/compositor/background_surface.c [hunk 1]
```diff
@@ -70,6 +75,7 @@ ws_buffer_type_id WS_OBJECT_TYPE_ID_IMAGE_BUFFER = {
     .get_height = get_height,
     .get_stride = get_stride,
     .get_format = get_format,
+    .get_bpp = get_bpp,
```
Comment (matthiasbeyer) [inline src/compositor/background_surface.c:78]: Was this member defined beforehand?

Comment (TheNeikos) [inline src/compositor/background_surface.c:78]: It wouldn't compile else (C99 doesn't allow forward declaration)

**Comment by neithernut** [comment]
ACK.

**Comment by matthiasbeyer** [comment]
> Mh... one of us has to implement the bpp getter for wl_wayland_buffers

Is this done? I guess this should go right into this PR as well!

**Comment by neithernut** [comment]
He added a preliminary implementation (he simply returns 4).
Considering that the buffer still needs some work (e.g. detecting whether it's really a `ws_shm_buffer`), this is acceptable for now.

**Comment by matthiasbeyer** [comment]
ACK.

**Pull request merged**